src/sssom/parsers.py

```diff
@@ -926,3 +915,34 @@ def split_dataframe_by_prefix(
             df_subset, prefix_map=dict(subconverter.bimap), meta=meta
         )
     return split_to_msdf
+
+
+def _ensure_valid_mapping_from_dict(mdict):
+    """
+    Return a valid mapping object if it can be constructed, else None.
+
+    :param mdict: A dictionary containing the mapping metadata.
+    :return: A valid Mapping object, or None.
+    """
+    try:
```

Comment:
this function could set the default mapping justification

```python
mdict[MAPPING_JUSTIFICATION] = MAPPING_JUSTIFICATION_UNSPECIFIED
```

and any other reasonable defaults

Reply:
Fantastic idea.. This will make this a bit more failsafe!